internal/core/operations/switch_active_version/switch_active_version_executor_test.go

```diff
@@ -0,0 +1,382 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package switch_active_version_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/topfreegames/maestro/internal/core/operations/switch_active_version"
+
+	"github.com/topfreegames/maestro/internal/core/services/interfaces"
+	mockeventsservice "github.com/topfreegames/maestro/internal/core/services/interfaces/mock/events_service"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	clock_mock "github.com/topfreegames/maestro/internal/adapters/clock/mock"
+	instance_storage_mock "github.com/topfreegames/maestro/internal/adapters/instance_storage/mock"
+	port_allocator_mock "github.com/topfreegames/maestro/internal/adapters/port_allocator/mock"
+	room_storage_mock "github.com/topfreegames/maestro/internal/adapters/room_storage/mock"
+	runtimemock "github.com/topfreegames/maestro/internal/adapters/runtime/mock"
+	schedulerstoragemock "github.com/topfreegames/maestro/internal/adapters/scheduler_storage/mock"
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	"github.com/topfreegames/maestro/internal/core/services/room_manager"
+	"github.com/topfreegames/maestro/internal/core/services/scheduler_manager"
+	"github.com/topfreegames/maestro/internal/validations"
+)
+
+// mockRoomAndSchedulerManager struct that holds all the mocks necessary for the
+// operation executor.
+type mockRoomAndSchedulerManager struct {
+	roomManager      *room_manager.RoomManager
+	schedulerManager *scheduler_manager.SchedulerManager
+	portAllocator    *port_allocator_mock.MockPortAllocator
+	roomStorage      *room_storage_mock.MockRoomStorage
+	instanceStorage  *instance_storage_mock.MockGameRoomInstanceStorage
+	runtime          *runtimemock.MockRuntime
+	eventsService    interfaces.EventsService
+	schedulerStorage *schedulerstoragemock.MockSchedulerStorage
+}
+
+// gameRoomIdMatcher matches the game room ID with the one provided.
+type gameRoomIdMatcher struct {
+	id string
+}
+
+// gameRoomVersionMatcher matches the game room version with the one provided.
+type gameRoomVersionMatcher struct {
+	version string
+}
+
+func TestSwitchActiveVersionOperation_Execute(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	mocks := newMockRoomAndSchedulerManager(mockCtrl)
+
+	err := validations.RegisterValidations()
+	if err != nil {
+		t.Errorf("unexpected error %d'", err)
+	}
+
+	currentVersion := "v1"
+
+	newScheduler := newValidScheduler()
+	newScheduler.PortRange.Start = 1000
+	newScheduler.MaxSurge = "3"
+
+	definition := &switch_active_version.SwitchActiveVersionDefinition{
+		NewActiveScheduler: newScheduler,
+	}
+
+	t.Run("should succeed - Execute switch active version operation", func(t *testing.T) {
```

Comment:
@caiordjesus WDYT about adding scenarios for when the `ListRoomsWithDeletionPriority` fails?

Reply:
@caiordjesus Also, one important scenario is when the game room creation fails during replacement, we don't want to end the operation with success when this happens, right?